tasks/main.yml

```diff
@@ -12,10 +12,29 @@
   apt: pkg=buildkite-agent state=present
 
 - name: Configure Buildkite
-  template: src=buildkite-agent.cfg.j2
-            dest="{{ buildkite_agent_conf_dir }}/buildkite-agent.cfg"
-            owner=buildkite-agent
-            group=buildkite-agent
-            mode=0600
+  template:
+    src: "buildkite-agent.cfg.v{{ buildkite_agent_major_version }}.j2"
+    dest: "{{ buildkite_agent_conf_dir }}/buildkite-agent.cfg"
+    owner: buildkite-agent
+    group: buildkite-agent
+    mode: "0600"
   notify:
     - Restart Buildkite
+
+- name: Turn off the installed service
+  service:
+    name: buildkite-agent
+    enabled: no
+    state: stopped
+
+- name: Create systemd template unit
+  copy:
+    src: /lib/systemd/system/buildkite-agent.service
+    dest: /lib/systemd/system/buildkite-agent@.service
+    owner: root
+    group: root
+    mode: "0644"
+
+- name: "Create buildkite-agent service instance #{{item}}"
+  systemd: name=buildkite-agent@{{item}} state=started enabled=yes
```

Comment:
In the meta for this role, it is defined as supporting Ubuntu 14.04 (Trusty). The `systemd` specific requirements in your PR make it so that it doesn't work on 14.04, but potentially does make it work on 16.04.

Can you please replace `trusty` in meta with `xenial`? Also, can you please update the Vagrant box reference in the repository `Vagrantfile` to use:

```ruby
config.vm.box = "bento/ubuntu-16.04"
```

Reply:
Hah, TIL. Thanks. Yes, updated.